pom.xml

```diff
@@ -40,7 +40,8 @@
     <project.build.resourceEncoding>UTF-8</project.build.resourceEncoding>
 
     <!-- default bytecode version for src/main -->
-    <main.java.version>1.7</main.java.version>
+    <!-- IntelliJ isn't clever enough to have different language levels for tests vs. main code -->
+    <main.java.version>1.8</main.java.version>
```

Comment:
Sorry, but this isn't a good enough reason. This will break api as it will make change the bytecode level for components that don't define it. Also, it changes the default, which was specifically there.

I do have a workaround, which I paste into the terminal any time dependencies change. That isn't terribly often, though certainly has been in the work you are doing.

```bash
find . -name '*.iml' -print0 | xargs -0 sed -i '' 's/JDK_1_6/JDK_1_8/g'
find . -name '*.iml' -print0 | xargs -0 sed -i '' 's/JDK_1_7/JDK_1_8/g'
sed -i '' 's/target="1.7"/target="1.8"/g' .idea/compiler.xml
sed -i '' 's/target="1.6"/target="1.8"/g' .idea/compiler.xml
```

ps please upvote https://youtrack.jetbrains.com/issue/IDEA-161060

https://github.com/openzipkin/zipkin/issues/1256#issuecomment-246564227